### ISSUE TITLE
would be nice to get absolute line numbers in warnings

### ISSUE THREAD

**Issue by Phillip-M-Feldman** [sphinx-doc/sphinx#6036]
When Sphinx produces a warning like the following, one must add two numbers to determine the absolute line number in the module:

docstring of mymath.find_root:212: WARNING: Block quote ends without a blank line; unexpected unindent.

It would be wonderful if there were an option to get absolute line numbers rather than line numbers relative to the start of the given doc string.

**Comment by tk0miya** [comment]
+1; resonable.

**Comment by artgoldberg** [comment]
I agree. In addition, I get Sphinx warnings in which the relative line number does not point to a line in the referenced Python element. E.g., in

    /root/host/Documents/obj_model/obj_model/core.py:docstring of obj_model.core.LocalAttribute:71: WARNING: Inline emphasis start-string without end-string.

`obj_model.core.LocalAttribute` does not have 71 lines (which means its docstring doesn't either).

I cannot debug this warning.

**Comment by zcorpan** [comment]
I ran into this, with this message
```
Warning, treated as error:
somefile.rst::Anonymous hyperlink mismatch: 1 references but 0 targets.
See "backrefs" attribute for IDs.
make: *** [html] Error 2
```
Without a line number I had to resort to debugging by binary-search removing content to find the error.

The mistake was this syntax error:
```rst
`link text <https://something.example/foo`__
```

A better message would be
```
somefile.rst:42:Syntax error in anonymous hyperlink: missing > character after URL.
```

**Comment by tk0miya** [comment]
@zcorpan Unfortunately, your case is not related with this issue. This is a problem caused by autodoc. BTW, I feel it is hard to warn your case better because Sphinx and docutils (reST parser) consider that the hyperlink does not have target URI (they searches the pair of angle brackets).

**Comment by tk0miya** [comment]
Note: It seems hard to fix this completely. Because the docstring might be modified by extensions (via `autodoc-process-docstring` event). For example, napoleon extension rewrites whole of docstring.

**Comment by ben-spiller** [comment]
+1 for this improvement, file-relative line numbers rather than docstring-relative numbers would make fixing doc warnings a lot easier :)

At the moment when I get a warning on docstring line 57 I first have to manually find the associated docstring in my file, then count down 57 lines (or use an editor that shows line numbers and do some mental arithmetic :D )

> Note: It seems hard to fix this completely. Because the docstring might be modified by extensions (via `autodoc-process-docstring` event). For example, napoleon extension rewrites whole of docstring.
True, but I think that problem also applies to the docstring-relative line numbers we're currently displaying. I think evitably users need be tolerant of the numbers occasionally being slightly off. 

However it'd still make a big different if the line number quoted was relative to start of file rather than start of docstring. Looks like this is possible e.g. https://stackoverflow.com/questions/11608747/extract-line-numbers-of-all-docstrings and/or maybe the ModuleAnalyzer can help (maybe for data members at any rate)

**Comment by umamibeef** [comment]
+1 for this
